React Data Fetching: ISS Tracker (#222)
neuefische/web-exercises: sessions/react-data-fetching/iss-tracker/README.md

# React Data Fetching: ISS Tracker

This challenge asks you to refactor the existing code of the ISS Tracker to use `SWR`.
You will need to fetch on interval and refetch on click of a button.

## Task

- Start the development server and make yourself familiar with the application.
- Switch to the [`App component`](./pages/index.js): it features a `coords` state, a `getISSCoords` function and a `useEffect` to fetch data on interval.

### Fetching on load

- Comment out all unnecessary code:

  - the `coords` state,
  - the `getISSCoords` function,
  - the `useEffect`,
  - for now: the `onRefresh` prop passed to the `Controls` component (you will adapt this in a minute).

- Implement `SWR` to fetch the same data.

  - `SWR` is already installed, so you just have to import it.
  - Create a loading and an error state as well.
  - Remember to adapt the `fetcher` function to create a proper `error` state [as explained in the SWR docs](https://swr.vercel.app/docs/error-handling#status-code-and-error-object).

>   - Note: `SWR` needs you to destructure a `data` object, but the `Map` and `Controls` component need a `coords` object passed to their props; how can you simply fix this naming issue without touching the props?

Reviewer: ```suggestion
  - Note: `SWR` needs you to destructure a `data` object, but the `Map` and `Controls` component need `longitude` and `latitude` as separate props; how can you simply pass the coordinates from `data` without touching the props?
```

Author: We have found a good compromise.

# React Data Fetching: ISS Tracker

This challenge asks you to refactor the existing code of the ISS Tracker to use `SWR`.
You will need to fetch on interval and refetch on click of a button.

## Task

- Start the development server and make yourself familiar with the application.
- Switch to the [`ISSTracker`](./components/ISSTracker/index.js) component: it features a `coords` state, a `getISSCoords` function and a `useEffect` to fetch data on interval.

### Fetching on load

- Comment out all unnecessary code:

  - the `coords` state,
  - the `getISSCoords` function,
  - the `useEffect`,
  - for now: the `onRefresh` prop passed to the `Controls` component (you will adapt this in a minute).

> - Implement `SWR` to fetch the same data.

Reviewer: ```suggestion
- Implement fetching of the same data with `SWR`.
```
